sourcegen-generator-java/src/test/java/io/micronaut/sourcegen/javapoet/write/FieldDefTest.java

```diff
@@ -32,4 +40,28 @@ public void twoDimensionalArrayField() throws IOException {
         assertEquals("byte[][] byteArray;", result);
     }
 
+
+    @Test public void annotatedGenericField() throws Exception {
+        PropertyDef.PropertyDefBuilder propertyDef = PropertyDef.builder("numbers");
+        var MIN_ANN = AnnotationDef.builder(ClassTypeDef.of("jakarta.validation.constraints.Min")).addMember("value", 1).build();
+        propertyDef.ofType(TypeDef.parameterized(ClassTypeDef.of(List.class), TypeDef.Primitive.FLOAT.wrapperType().annotated(MIN_ANN)));
+
+        RecordDef.RecordDefBuilder builder = RecordDef.builder("Record").addProperty(propertyDef.build());
+        JavaPoetSourceGenerator generator = new JavaPoetSourceGenerator();
+        String result;
+        try (StringWriter writer = new StringWriter()) {
+            generator.write(builder.build(), writer);
+            result = writer.toString();
+        }
+
+        assertThat(result).isEqualTo("import jakarta.validation.constraints.Min;\n" +
+            "import java.lang.Float;\n" +
+            "import java.util.List;\n" +
+            "\n" +
+            "record Record(\n" +
+            "    List<@Min(1) Float> numbers\n" +
+            ") {\n" +
+            "}\n");
+    }
+
```

Comment:
Can you test if 
```java
propertyDef.ofType(TypeDef.Primitive.FLOAT.wrapperType().annotated(MIN_ANN))
```
would work correctly? Before the change annotatons would be added on properties, but since they can be on types now, we should check if both cases work.

Reply:
It does work 👍